opengever/base/error_log.py

```diff
@@ -0,0 +1,105 @@
+from copy import deepcopy
+from datetime import datetime
+from opengever.base.interfaces import IBaseSettings
+from opengever.base.utils import to_safe_html
+from opengever.redis.client import redis_client_manager
+from plone import api
+import json
+
+
+def is_user_visible_error_logs_feature_enabled():
+    return api.portal.get_registry_record(
+        'is_user_visible_error_logs_feature_enabled', interface=IBaseSettings)
+
+
+class ErrorLogRedis:
+    STORAGE_KEY = 'error_log'
+    MAX_LOG_ITEMS = 200
+
+    def __init__(self, redis_client):
+        self.client = redis_client
+
+    def push(self, error_log_item):
+        if not isinstance(error_log_item, ErrorLogItem):
+            raise Exception('Requires an ErrorLogItem')
+
+        self.client.lpush(self.STORAGE_KEY, error_log_item.to_json())
+        self.trim()
+
+    def list_all(self):
+        for value in self.client.lrange(self.STORAGE_KEY, 0, self.MAX_LOG_ITEMS):
+            yield ErrorLogItem.from_json(value)
+
+    def list_all_by_userid(self, userid):
+        for item in self.list_all():
+            if item.userid == userid:
+                yield item
+
+    def trim(self):
+        self.client.ltrim(self.STORAGE_KEY, 0, self.MAX_LOG_ITEMS - 1)
+
+    def clear(self):
+        self.client.delete(self.STORAGE_KEY)
+
+
+class DisabledErrorLogRedis:
```

Comment:
```suggestion
class NullErrorLog:
```
Ist nicht wirklich ein Redis Error Log.